packages/swirl-components/src/components/swirl-popover-trigger/swirl-popover-trigger.tsx

```diff
@@ -0,0 +1,105 @@
+import { Component, Element, h, Host, Prop, Watch } from "@stencil/core";
+
+@Component({
+  shadow: false,
+  scoped: true,
+  styleUrl: "swirl-popover-trigger.css",
+  tag: "swirl-popover-trigger",
+})
+export class SwirlPopoverTrigger {
+  @Element() el!: HTMLSwirlPopoverTriggerElement;
+
+  @Prop() popover!: string | HTMLSwirlPopoverElement;
+  @Prop() setAriaAttributes?: boolean = true;
+
+  private popoverOpen: boolean = false;
+
+  componentDidLoad() {
+    this.updateTriggerElAriaAttributes();
+  }
+
+  @Watch("popover")
+  watchPopover() {
+    this.updateTriggerElAriaAttributes();
+  }
+
+  private getPopoverEl() {
+    return typeof this.popover === "string"
+      ? document.querySelector<HTMLSwirlPopoverElement>(`#${this.popover}`)
+      : this.popover;
+  }
+
+  private getTriggerEl() {
+    return this.el.children[0] as HTMLElement;
```

Comment:
**question (non-blocking):** how can we ensure there is only one child? can we throw somekind of error through stencil? maybe in the componentDidLoad function?

Reply:
Done.